sdk/test/resource/resource_test.cc

```diff
@@ -17,30 +18,29 @@
 #  define putenv _putenv
 #endif
 
-class TestResource : public opentelemetry::sdk::resource::Resource
+using namespace opentelemetry::sdk::resource;
+
+class TestResource : public Resource
 {
 public:
-  TestResource(opentelemetry::sdk::resource::ResourceAttributes attributes =
-                   opentelemetry::sdk::resource::ResourceAttributes())
-      : Resource(attributes)
-  {}
+  TestResource(ResourceAttributes attributes = ResourceAttributes()) : Resource(attributes) {}
 };
 
 TEST(ResourceTest, create_without_servicename)
 {
 
-  opentelemetry::sdk::resource::ResourceAttributes expected_attributes = {
+  ResourceAttributes expected_attributes = {
       {"service", "backend"},
       {"version", (uint32_t)1},
       {"cost", 234.23},
-      {"telemetry.sdk.language", "cpp"},
-      {"telemetry.sdk.name", "opentelemetry"},
-      {"telemetry.sdk.version", OPENTELEMETRY_SDK_VERSION},
-      {"service.name", "unknown_service"}};
+      {SemanticConventions::GetAttributeTelemetrySdkLanguage(), "cpp"},
```

Comment:
I honestly would prefer an API model similar to Qt5 "translation resources", which solves the same problem - just in a different knowledge domain. More on that [here](https://doc.qt.io/qt-5/qtlinguist-hellotr-example.html)

To elaborate:

- let's say you have a namespace called `MyAttributeNames` , and either `constexpr` or `extern` that looks like Qt5 `tr(name)` ... Let's name it `attr(name)`

- then in order to obtain the actual name for a given system, instead of `SemanticConventions::GetAttributeTelemetrySdkLanguage()` , you'd write `attr("sdk.language").... Moreover, since you are not gonna be using all attributes at once, it is Okay to declare these as literals, e.g. OTEL_ATTR_SDK_LANGUAGE

Then instead of having N setters, which will also present a challenge - should you ever need to project this C++ code to C#, Python or Javascript/Node, etc... you'd have ONE method - called `AttributeNames::attr` and you'd invoke it as:

```cpp
using namespace AttributeNames;
...
{attr(OTEL_ATTR_SDK_LANGUAGE), "cpp"}
... // or
{attr("sdk.language", "cpp")}
```

Then we should strive the transform to be done at compile time, without generating those 50 methods. It does not scale. It'd scale nicely if you have one method / one utility function, hopefully `constexpr`. And a table of literals, `resources` in a classic definition of this world in C++ (as in translation resources, resource compiler, etc.).

Again, `tr()` from Qt5 seems like the most reasonable way to achieve that.